api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java

```diff
@@ -0,0 +1,861 @@
+        assertEquals(encounter3.getLocation(),location1);
+        
+        Encounter encounter4 = encounterService.getEncounter(4);
+        assertNotNull(encounter4);
+        assertEquals(encounter4.getLocation(),location1);
+        
+        Encounter encounter5 = encounterService.getEncounter(5);
+        assertNull(encounter5);
+        
+        Encounter encounter6 = encounterService.getEncounter(6);
+        assertNull(encounter6);
+        
+        Encounter encounter7 = encounterService.getEncounter(7);
+        assertNull(encounter7);
+        
+        Encounter encounter8 = encounterService.getEncounter(8);
+        assertNull(encounter8);
+    }
+    
+    @Test
+    public void getEncounterByUuid_adminCanAccessEncounterFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+        
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+        
+        Encounter encounter1 = encounterService.getEncounterByUuid("7519d653-393b-4118-9c83-a3715b82d4ac");
+        assertNotNull(encounter1);
+        assertEquals(encounter1.getLocation(),location1);
+        
+        Encounter encounter2 = encounterService.getEncounterByUuid("8519d653-393b-4118-9c83-a3715b82d4ac");
+        assertNotNull(encounter2);
+        assertEquals(encounter2.getLocation(),location2);
+        
+        Encounter encounter3 = encounterService.getEncounterByUuid("6519d653-393b-4118-9c83-a3715b82d4ac");
+        assertNotNull(encounter3);
+        assertEquals(encounter3.getLocation(),location1);
+        
+        Encounter encounter4 = encounterService.getEncounterByUuid("eec646cb-c847-45a7-98bc-91c8c4f70add");
+        assertNotNull(encounter4);
+        assertEquals(encounter4.getLocation(),location1);
+        
+        Encounter encounter5 = encounterService.getEncounterByUuid("e403fafb-e5e4-42d0-9d11-4f52e89d148c");
+        assertNotNull(encounter5);
+        assertEquals(encounter5.getLocation(),location2);
+        
+        Encounter encounter6 = encounterService.getEncounterByUuid("y403fafb-e5e4-42d0-9d11-4f52e89d123r");
+        assertNotNull(encounter6);
+        assertEquals(encounter6.getLocation(),location2);
+        
+        Encounter encounter7 = encounterService.getEncounterByUuid("9519d653-393b-4118-9c83-a3715b82d4ac");
+        assertNotNull(encounter7);
+        assertEquals(encounter7.getLocation(),location2);
+        
+        Encounter encounter8 = encounterService.getEncounterByUuid("0619d653-393b-4118-9c83-a3715b82d4ac");
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncounterByUuid_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){

```

Comment:
Have you checked for null user property?

Reply:
Fixed 😃